src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpersTests.cs

```diff
@@ -0,0 +1,1171 @@
+    {
+        get
+        {
+            if (name is null)
+                throw new ArgumentNullException(nameof(name));
+            return name;
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task ArgumentNullExceptionThrowIfNull_FixAppliesAsAppropriate()
+        {
+            await new VerifyCS.Test()
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestCode =
+@"
+using System;
+
+namespace System
+{
+    public class ArgumentNullException : Exception
+    {
+        public ArgumentNullException() { }
+        public ArgumentNullException(string paramName) { }
+        public ArgumentNullException(string paramName, string message) { }
+        public ArgumentNullException(string paramName, Exception innerException) { }
+        public static void ThrowIfNull(object argument, string name = null) { }           
+    }
+}
+
+class C
+{
+    void M(string arg)
+    {
+        {|CA1510:if (arg is null)
+            throw new ArgumentNullException(nameof(arg));|}
+        {|CA1510:if (arg == null)
+            throw new ArgumentNullException(nameof(arg), (string)null);|}
+        {|CA1510:if (null == arg)
+            throw new ArgumentNullException(nameof(arg));|}
+        {|CA1510:if (arg is null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }|}
+        {|CA1510:if (arg == null)
+        {
+            throw new ArgumentNullException(nameof(arg), """");
+        }|}
+        {|CA1510:if (null == arg)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }|}
+        {|CA1510:if (null == arg)
+        {
+            throw new ArgumentNullException(""something else"");
+        }|}
+
+        if (arg == ""test"")
+        {
+            Console.WriteLine(arg);
+        }
+        else {|CA1510:if (arg is null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }|}
+
+        if (arg is null)
+            throw new ArgumentNullException(nameof(arg), ""possibly meaningful message"");
+
+        if (arg is null)
+            throw new ArgumentNullException(nameof(arg), new Exception());
+
+        if (arg is null)
+        {
+            Console.WriteLine(); // another operation in the block
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (arg is not null) // inverted condition
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (arg != null) // inverted condition
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (null != arg) // inverted condition
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (arg is null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+        else if (arg == ""test"")
+        {
+            Console.WriteLine(arg);
+        }
```

Comment:
Wonder why this is not covered, seems it could be fixed with
```cs
    ArgumentNullException.ThrowIfNull(arg);
    if (arg == ""test"")
    {
        Console.WriteLine(arg);
    }
```

Reply:
It's just a somewhat rare construction and didn't seem worth handling.  It also wasn't clear to me if there was a trivial way to handle the fix when there was an else part like this (without special-casing it, the whole else gets deleted when replacing the if).  If there's a simple way to handle this, we can certainly add it in.